test/scalacheck/scala/collection/immutable/ListMapBuilderProperties.scala

```diff
@@ -0,0 +1,46 @@
+package scala.collection.immutable
+
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop._
+import org.scalacheck._
+import org.scalacheck.Gen._
+
+object ListMapBuilderProperties extends Properties("immutable.ListMapBuilder") {
+
+  type K = Int
+  type V = Int
+
+  sealed trait Transform
+
+  object Transform {
+    case class AddOne(kv: (K, V)) extends Transform
+    case class AddAllList(kvs: List[(K, V)]) extends Transform
+    case class AddAllMap(kvs: Map[K, V]) extends Transform
+    case object Clear extends Transform
+  }
+
+
+
+  val transformGen: Gen[List[Transform]] =
+    listOf(oneOf(
```

Comment:
By default, `oneOf` would generate each of the elements with equal frequency. Instead, we should use [`Gen.frequency`](https://github.com/rickynils/scalacheck/blob/master/src/main/scala/org/scalacheck/Gen.scala#L550) so as to, for instance, make `clear` less frequent than all the others?

Reply:
This is a question about test coverage, which is rather too deep. Good coverage would depend on how complex is the inner state (mutable or not) of the data structure. Without knowing that, it is more like a hunch. You would also need to consider how long are the lists of elements in the `addAll` method. 

Another test case more specialised for an immutable  `ListMap`, in particular, would be to check what happens when you insert a same key twice.

In any case, the property test as it is, and the PR, is good enough to merge.